shell/common/layer_tree_holder.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_COMMON_LAYER_TREE_HOLDER_H_
+#define FLUTTER_SHELL_COMMON_LAYER_TREE_HOLDER_H_
+
+#include <memory>
+
+#include "flow/layers/layer_tree.h"
+
+namespace flutter {
+
+class LayerTreeHolder {
+ public:
+  LayerTreeHolder() = default;
+
+  ~LayerTreeHolder() = default;
+
+  bool IsEmpty() const;
+
+  std::unique_ptr<LayerTree> Get();
```

Comment:
How about `PushIfNewer` and `Pop`? `Get` implies a getter that is idempotent.

Reply:
Good idea.